blocks/store/reducer.js

```diff
@@ -43,6 +43,28 @@ export function blockTypes( state = {}, action ) {
 	return state;
 }
 
+/**
+ * Reducer managing the block type implementations
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function implementations( state = {}, action ) {
```

Comment:
I don't necessarily oppose, but to me it was unexpected to see this as a separate branch on the client. Does the client really care about which bits of information are part of the definition and which of the implementation? As long as we have distinct actions, we can treat the bits differently as they come in if need be, but then we store everything in a common place for a single block type.